### ISSUE TITLE
stream ID X; INTERNAL_ERROR

### ISSUE THREAD

**Issue by danielma82** [prometheus-operator/kube-prometheus#411]
**What happened?**
Running the new setup command results in stream ID INTERNAL_ERRORs for the `kubectl create -f manifests` step.

Unable to connect to the server: stream error: stream ID 1; INTERNAL_ERROR
Tue 11 Feb 2020 11:02:32 GMT

Unable to connect to the server: stream error: stream ID 1; INTERNAL_ERROR
Tue 11 Feb 2020 11:02:33 GMT

Unable to connect to the server: stream error: stream ID 1; INTERNAL_ERROR
Tue 11 Feb 2020 11:02:34 GMT

Error encountered for both 
```
kubectl create -f manifests/setup                                                                                                  
until kubectl get servicemonitors --all-namespaces ; do date; sleep 1; echo ""; done
kubectl create -f manifests/ 

```
and running `kubectl create -f manifests/setup` and `kubectl create -f manifests/` separately.

Different output when running `kubectl create -f manifests` on its own (`kubectl create -f manifests/setup` runs successfully):
```
error when creating "manifests/alertmanager-alertmanager.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/alertmanagers: stream error: stream ID 73; INTERNAL_ERROR
error when creating "manifests/alertmanager-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 81; INTERNAL_ERROR
error when creating "manifests/couchbase-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/default/servicemonitors: stream error: stream ID 83; INTERNAL_ERROR
error when creating "manifests/grafana-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 145; INTERNAL_ERROR
error when creating "manifests/kube-state-metrics-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 161; INTERNAL_ERROR
error when creating "manifests/node-exporter-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 173; INTERNAL_ERROR
error when creating "manifests/prometheus-operator-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 201; INTERNAL_ERROR
error when creating "manifests/prometheus-prometheus.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/prometheuses: stream error: stream ID 203; INTERNAL_ERROR
error when creating "manifests/prometheus-rules.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/prometheusrules: stream error: stream ID 221; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitor.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 227; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitorApiserver.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 229; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitorCoreDNS.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 231; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitorKubeControllerManager.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 233; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitorKubeScheduler.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 235; INTERNAL_ERROR
error when creating "manifests/prometheus-serviceMonitorKubelet.yaml": Post https://192.168.64.6:8443/apis/monitoring.coreos.com/v1/namespaces/monitoring/servicemonitors: stream error: stream ID 237; INTERNAL_ERROR
```

but still the same error.

When running the delete command this returns:
`Delete https://192.168.64.5:8443/api/v1/namespaces/monitoring/serviceaccounts/prometheus-operator: dial tcp 192.168.64.5:8443: connect: connection refused
`
And then the apiserver goes down:
host: Running
kubelet: Running
apiserver: Stopped
kubeconfig: Configured

**Did you expect to see some different?**
For it to work. Way back before this setup folder was required I was able to start kube-prometheus with no problems, with `kubectl apply -f manifests/`

**How to reproduce it (as minimally and precisely as possible)**:
Start Minikube, either normally (`minikube start`) or with the command specified in the README.
Run the command in the README or run separately.
Error.

**Environment**
minikube version: v1.7.2

Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:34 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:28:45 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc9
  GitCommit:        d736ef14f0288d6993a1845745d6756cfc9ddd5a
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

* Prometheus Operator version:
 Image:      quay.io/coreos/prometheus-operator:v0.35.1

* Kubernetes version information:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-23T14:21:54Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:27:17Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:
minikube 

* Manifests:
Unsure

* Prometheus Operator Logs:
Deployment does not start

* Prometheus Logs:
Unsure how to get

**Anything else we need to know?**:

**Comment by olegmayko** [comment]
can you check/provide output of cmd?
> minikube logs

**Comment by czedyousfi** [comment]
I had exactly the same problem. Using the branch release-0.3 resolved my issue.

**Comment by danielma82** [comment]
Sorry for the late response. 
Funnily enough I tried this again and actually the manifests/setup and manifests all setup without the same error, but the next problem was that the prometheus-k8s-0 pod I would expect to come up to get `kubectl --namespace monitoring port-forward svc/prometheus-k8s 9090` working does not appear.

So I deleted and restarted minikube using the provided command in the README, and then ran the create command in the README and the stream ID error reappeared.

Attached are the minikube logs.

[minikube.log](https://github.com/coreos/kube-prometheus/files/4214331/minikube.log)
